code/datums/memory/tattoo_kit.dm

```diff
@@ -17,17 +27,18 @@
 		. += span_notice("This kit has enough ink for [uses] use\s.")
 	. += span_boldnotice("You can use a toner cartridge to refill this.")
 
-/obj/item/tattoo_kit/attackby(obj/item/toner/ink_cart, mob/living/tattoo_artist, list/modifiers, list/attack_modifiers)
-	. = ..()
+/obj/item/tattoo_kit/item_interaction(mob/living/user, obj/item/toner/ink_cart, list/modifiers)
 	if(!istype(ink_cart))
-		return
+		return NONE
 	var/added_amount = round(ink_cart.charges / 5)
 	if(added_amount == 0)
-		return
+		balloon_alert(user, "none left!")
+		return ITEM_INTERACT_BLOCKING
 	added_amount = min(uses + added_amount, max_uses)
 	uses += min(max_uses, added_amount)
 	qdel(ink_cart)
```

Comment:
Probably should also be preventing refills if the kit is already full, not sure why this wasn't the case before but it should be.